django/db/backends/sqlite3/base.py

```diff
@@ -451,17 +451,19 @@ def _sqlite_date_trunc(lookup_type, dt, tzname, conn_tzname):
     if dt is None:
         return None
     if lookup_type == 'year':
-        return "%i-01-01" % dt.year
+        return f'{dt.year:04d}-01-01'
```

Comment:
Can we just change `%i` to `%04i` for the years in this fix.

Leave the f-string optimisations to the original PR because those changes apply to `_sqlite_time_trunc()` which have been split from these. In addition, the added exceptions need a little test.

Reply:
Fair enough, done.

Leaving the f-strings and `NotImplementError` pathways for the optimization PR.